src/wallet/wallet.cpp

```diff
@@ -2754,6 +2771,11 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     }
                     break; // Done, enough fee included.
                 }
+                else if (!pick_new_inputs) {
+                    // This shouldn't happen
```

Comment:
yes, i don't think that can happen (a change in fee estimation) but belts and suspenders.
In any case will be cleaned up post 0.15 with effective value logic and only calculating the needed fee rate once.